packages/jitm/src/class-pre-connection-jitm.php

```diff
@@ -0,0 +1,143 @@
+				'button_caption' => __( 'Setup Jetpack', 'jetpack' ),
+			),
+			array(
+				'id'             => 'jpsetup-upload',
+				'message_path'   => '/wp:upload:admin_notices/',
+				'message'        => __( 'Do you want lightning-fast images?', 'jetpack' ),
+				'description'    => __( 'Setup Jetpack, enable Site Accelerator, and start serving your images lightning fast, for free.', 'jetpack' ),
+				'button_link'    => esc_url( \Jetpack::init()->build_connect_url( true, false, 'pre-connection-jitm-upload' ) ),
+				'button_caption' => __( 'Setup Jetpack', 'jetpack' ),
+			),
+			array(
+				'id'             => 'jpsetup-widgets',
+				'message_path'   => '/wp:widgets:admin_notices/',
+				'message'        => __( 'Looking for even more widgets?', 'jetpack' ),
+				'description'    => __( 'Setup Jetpack for great additional widgets like business hours and maps.', 'jetpack' ),
+				'button_link'    => esc_url( \Jetpack::init()->build_connect_url( true, false, 'pre-connection-jitm-widgets' ) ),
+				'button_caption' => __( 'Setup Jetpack', 'jetpack' ),
+			),
+		);
+	}
+
+	/**
+	 * Filters and formats the messages for the client-side JS renderer
+	 *
+	 * @param string $message_path Current message path.
+	 *
+	 * @return array Formatted messages.
+	 */
+	private function filter_messages( $message_path ) {
+		$messages = $this->get_raw_messages();
+
+		$formatted_messages = array();
+
+		foreach ( $messages as $message ) {
+			if ( ! preg_match( $message['message_path'], $message_path ) ) {
+				continue;
+			}
+
+			$obj                 = new \stdClass();
+			$obj->CTA            = array( // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				'message'   => $message['button_caption'],
+				'newWindow' => false,
+			);
+			$obj->url            = $message['button_link'];
+			$obj->id             = $message['id'];
+			$obj->is_dismissible = true;
+			$obj->content        = array(
+				'message'     => $message['message'],
+				'description' => $message['description'],
+				'list'        => array(),
+				'icon'        => 'jetpack',
+			);
+
+			$formatted_messages[] = $obj;
+		}
+
+		return $formatted_messages;
+	}
+
+	/**
+	 * Retrieve the current message to display keyed on query string and message path
+	 *
+	 * @param string $message_path The message path to ask for.
+	 * @param string $query The query string originally from the front end. Unused in this subclass.
+	 * @param bool   $full_jp_logo_exists If there is a full Jetpack logo already on the page.
+	 *
+	 * @return array The JITMs to show, or an empty array if there is nothing to show
+	 */
+	public function get_messages( $message_path, $query, $full_jp_logo_exists ) {
+		/**
+		 * This is an experiment for partners to test. Allow customization of the behavior of pre-connection banners.
+		 *
+		 * @since 8.6.0
+		 *
+		 * @param bool $enable_pre_connection_jitms Should pre-connection JITMs appear? Default to false.
+		 */
```

Comment:
This can be replaced since the parser is already picking up the docblock from another file:

```suggestion
		/** This filter is documented in  class.jetpack-connection-banner.php */
```